collector/src/test/java/io/prometheus/jmx/SnakeCaseAttrTest.java

```diff
@@ -0,0 +1,41 @@
+package io.prometheus.jmx;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.Arrays;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@RunWith(Parameterized.class)
+public class SnakeCaseAttrTest {
+    @Parameterized.Parameters(name = "{index}: testAttrToSnakeAndLowerCase(expected={0} actual={1}")
+    public static Iterable<Object[]> data() {
+        return Arrays.asList(new Object[][] {
+                { "test_test", "testTest" }, { "test_test_test", "testTestTest" }, {"test_test", "test_test"},
+                { "test1", "test1"}, { "start_time_$1_$2", "StartTime_$1_$2" }, { "a", "A" }, { "a_a", "AA" },
+                { null, null },  { "", "" }, { " ", " " }, { "test_test\n_test", "testTest\nTest" }
+        });
+    }
+
+    private final String expected;
+    private final String input;
+
+    public SnakeCaseAttrTest(String expected, String input) {
+        this.expected = expected;
+        this.input = input;
+    }
+
+    @Test
+    public void testAttrToSnakeAndLowerCase() {
+        JmxCollector.Receiver rc = mock(JmxCollector.Receiver.class);
```

Comment:
I can't use the constructor because I can't access it via non-static context.

Reply:
Why not make the function static?

Reply:
I can solve it that way if you're OK with it. Another way to solve it is to pull it out into its own file.